Add entry points for supported inventory sources

jessicamack opened the pull request:
A related change for https://github.com/ansible/awx/pull/15658

jessicamack committed:
add entry points for supported inventory sources

also fix inventory name to match class

webknjaz commented on pyproject.toml:
```diff
@@ -89,6 +89,12 @@ insights = "awx_plugins.inventory.plugins:insights"
 openshift_virtualization = "awx_plugins.inventory.plugins:openshift_virtualization"
 constructed = "awx_plugins.inventory.plugins:constructed"
 
+[project.entry-points."awx_plugins.inventory.supported"]  # new entry points group name
```
```suggestion
[project.entry-points."awx_plugins.inventory.supported"]  # plugins that awx will skip activating automatically
```

webknjaz commented on pyproject.toml:
```diff
@@ -76,7 +76,7 @@ thycotic_tss = "awx_plugins.credentials.tss:tss_plugin"
 
 [project.entry-points."awx_plugins.inventory"]  # new entry points group name
-azure-rm = "awx_plugins.inventory.plugins:azure_rm"
+azure_rm = "awx_plugins.inventory.plugins:azure_rm"
```
The tests must be adjusted for this, so they don't fail.

jessicamack replied:
Which tests? When I point the requirements for awx at this branch on my repo the related tests pass. I'll run them again to confirm.

webknjaz replied:
Ones failing in CI — smoke: https://github.com/ansible/awx-plugins/blob/8e8d03b/tests/importable_test.py#L76 / https://github.com/ansible/awx-plugins/actions/runs/12121157958/job/33791551145?pr=61#step:17:51

webknjaz commented on pyproject.toml:
```diff
@@ -89,6 +89,12 @@ insights = "awx_plugins.inventory.plugins:insights"
 constructed = "awx_plugins.inventory.plugins:constructed"
 
+[project.entry-points."awx_plugins.inventory.supported"]  # new entry points group name
+satellite6 = "awx_plugins.inventory.plugins:satellite6_supported"
+controller = "awx_plugins.inventory.plugins:controller_supported"
+insights = "awx_plugins.inventory.plugins:insights_supported"
+openshift_virtualization = "awx_plugins.inventory.plugins:openshift_virtualization_supported"
```
These should probably be mentioned in the tests too.

jessicamack replied:
Added a test to cover the supported inventory plugins

jessicamack committed:
make comment more accurate

Co-authored-by: 🇺🇦 Sviatoslav Sydorenko (Святослав Сидоренко) <wk.cvs.github@sydorenko.org.ua>

jessicamack committed:
update plugin name

jessicamack committed:
add tests for supported inventory plugins

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

webknjaz commented on tests/importable_test.py:
```diff
@@ -136,6 +136,29 @@ def __str__(self) -> str:
     ),
 )
 
+supported_inventory_plugins = (
```
Merge them. From the runtime perspective, they are the same and there shouldn't be any difference in this.
```suggestion
```

webknjaz commented on tests/importable_test.py:
```diff
@@ -192,3 +221,14 @@ def test_entry_points_are_inventory_plugin(
     loaded_plugin_class = entry_points[entry_point.name].load()
 
     assert issubclass(loaded_plugin_class, PluginFileInjector)
+
+
+@with_supported_inventory_plugins
+def test_entry_points_are_supported_inventory_plugin(
+        entry_point: EntryPointParam,
+) -> None:
+    """Ensure all exposed inventory plugins are of the same class."""  # noqa: D200, DAR101; FIXME
+    entry_points = _discover_entry_points(group=entry_point.group)
+    loaded_plugin_class = entry_points[entry_point.name].load()
+
+    assert issubclass(loaded_plugin_class, PluginFileInjector)
```
```suggestion
```

webknjaz commented on tests/importable_test.py:
```diff
@@ -150,10 +173,16 @@ def __str__(self) -> str:
 with_all_plugins = pytest.mark.parametrize(
     'entry_point',
-    credential_plugins + inventory_plugins,
+    credential_plugins + inventory_plugins + supported_inventory_plugins,
```
```suggestion
    credential_plugins + inventory_plugins,
```

webknjaz commented on tests/importable_test.py:
```diff
@@ -150,10 +173,16 @@ def __str__(self) -> str:
     ids=str,
 )
 
+with_supported_inventory_plugins = pytest.mark.parametrize(
+    'entry_point',
+    supported_inventory_plugins,
+    ids=str,
+)
+
```
```suggestion
```

webknjaz reviewed (commented):
I suggest simplifying the tests like this.

jessicamack committed:
add rhv to supported inventory plugins

jessicamack committed:
Merge branch 'add-entry-points' of https://github.com/jessicamack/awx-plugins into add-entry-points

jessicamack committed:
simplify test

From the runtime perspective, they are the same so there's no reason to separate
